src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -74,42 +80,68 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             if (declarationContext is VariableDeclarationSyntax varDecl)
             {
                 typeSyntax = varDecl.Type;
+                variableIdentifier = varDecl.Variables.First().Identifier;
             }
             else if (declarationContext is ForEachStatementSyntax forEach)
             {
                 typeSyntax = forEach.Type;
+                variableIdentifier = forEach.Identifier;
             }
             else if (declarationContext is DeclarationExpressionSyntax declarationExpression)
             {
                 typeSyntax = declarationExpression.Type;
-                if (declarationExpression.Designation.IsKind(SyntaxKind.ParenthesizedVariableDesignation, out ParenthesizedVariableDesignationSyntax variableDesignation))
+                if (declarationExpression.Designation.IsKind(SyntaxKind.ParenthesizedVariableDesignation, out ParenthesizedVariableDesignationSyntax? variableDesignation))
                 {
                     parensDesignation = variableDesignation;
                 }
             }
             else
             {
-                throw ExceptionUtilities.UnexpectedValue(declarationContext.Kind());
+                throw ExceptionUtilities.UnexpectedValue(declarationContext?.Kind());
             }
 
             if (parensDesignation is null)
             {
                 typeSyntax = typeSyntax.StripRefIfNeeded();
 
+                var newTypeSymbol = semanticModel.GetTypeInfo(typeSyntax, cancellationToken).ConvertedType;
```

Comment:
`var convertedType`?

also, how do you know it's not null?

Reply:
It has been not null since the introduction of the feature. I'm not sure how to take current assumed non-null other than add a Debug assert like I did. The code prior to this change would null deref if it ever was null

Reply:
would still call thsi `var convertedType`.    Am still nervous about the null.  :-/